src/bun.js/base.zig

```diff
@@ -94,6 +101,11 @@ pub fn toJS(globalObject: *JSC.JSGlobalObject, comptime ValueType: type, value:
                 return value.toJS(globalObject);
             }
 
+            // must come after toJS check in case this enum implements its own serializer.
```

Comment:
What do you suggest it should be instead, if anything?

Reply:
an error asking the caller to specify what they meant, instead of implicitly choosing for them.

Reply:
Erroring here prevents comptime marshaling of structs with enum fields

This should work out of the box. It will not work without automatic enum serialization:
```zig
const Foo = enum { a, b };
const Bar = struct {
  foo: Foo,
  bar: []const u8,
};
const zig: Foo = .{ ... };
const js = JSC.toJS(global, Bar, zig, .allocated);
```

Reply:
imo its ambiguous if it means `{ bar: "...", foo: "a" }` or `{ bar: "...", foo: 0 }`. it's good the comment describes this default, but my issue is the fact that default is chosen. it will probably cause a bug down the line.

is this used right now? if so i wont block the pr on it, but if it isn't please remove it.

Reply:
It's being used to serialize `Diff` structs.